# Assignment 1
### Understanding Uncertainty
### Due 9/5

1. Create a new public repo on Github under your account. Include a readme file.
2. Clone it to your machine. Put this file into that repo.
3. Use the following function to download the example data for the course:

In [ ]:
def download_data(force=False):
    """Download and extract course data from Zenodo."""
    import urllib.request, zipfile, os
    
    zip_path = 'data.zip'
    data_dir = 'data'
    
    if not os.path.exists(zip_path) or force:
        print("Downloading course data")
        urllib.request.urlretrieve(
            'https://zenodo.org/records/16954427/files/data.zip?download=1',
            zip_path
        )
        print("Download complete")
    else:
        print("Download file already exists")
        
    if not os.path.exists(data_dir) or force:
        print("Extracting data files...")
        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
            zip_ref.extractall(data_dir)
        print("Data extracted")
    else:
        print("Data directory already exists")

download_data()

Download file already exists
Data directory already exists


In [3]:
import pandas as pd
import numpy as np



In [4]:
iowa_liquor=pd.read_csv("iowa.csv")
iowa_liquor

Invoice/Item Number        Date  Store Number  \
0          INV-59108400026  06/06/2023          3723   
1             S16879800057  01/16/2014          3926   
2          INV-05301100019  06/05/2017          3829   
3          INV-40973500083  10/14/2021          5102   
4          INV-17022500013  01/18/2019          2560   
...                    ...         ...           ...   
159899     INV-20689000058  07/17/2019          2565   
159900        S05071400061  04/16/2012          3583   
159901     INV-22880300009  10/29/2019          3896   
159902     INV-58180900067  05/05/2023          4057   
159903     INV-64335600006  11/16/2023          5371   

                                         Store Name Zip Code  \
0                                J D SPIRITS LIQUOR    51040   
1                       LIQUOR DOWNTOWN / IOWA CITY    52240   
2                          GARY'S FOODS / MT VERNON    52314   
3                                    WILKIE LIQUORS    52314   
4                        HY-VEE FOOD STORE / MARION    52302   
...                                             ...      ...   
159899            HY-VEE FOOD STORE #1636 / SPENCER    51301   
159900                      MAC'S LIQUOR AND VIDEOS    50060   
159901  SMOKIN' JOE'S #18 TOBACCO AND LIQUOR OUTLET    52240   
159902                       TEQUILA'S LIQUOR STORE    50316   
159903                              STATE FOOD MART    52722   

                         Category Name                       Vendor Name  \
0                STRAIGHT RYE WHISKIES                  INFINIUM SPIRITS   
1                       VODKA 80 PROOF                HEAVEN HILL BRANDS   
2                    CANADIAN WHISKIES                   DIAGEO AMERICAS   
3                    AMERICAN SCHNAPPS                   JIM BEAM BRANDS   
4                      WHISKEY LIQUEUR              SAZERAC COMPANY  INC   
...                                ...                               ...   
159899    AMERICAN CORDIALS & LIQUEURS                         LUXCO INC   
159900                 COFFEE LIQUEURS  PERNOD RICARD USA/AUSTIN NICHOLS   
159901               IMPORTED DRY GINS                   DIAGEO AMERICAS   
159902  NEUTRAL GRAIN SPIRITS FLAVORED          OLE SMOKY DISTILLERY LLC   
159903       STRAIGHT BOURBON WHISKIES                   JIM BEAM BRANDS   

        Item Number                          Item Description  \
0             27102                         TEMPLETON RYE 4YR   
1             35416                    BURNETT'S VODKA 80 PRF   
2             11296                               CROWN ROYAL   
3             82787                      DEKUYPER BUTTERSHOTS   
4             64863                 FIREBALL CINNAMON WHISKEY   
...             ...                                       ...   
159899        76488                        PARAMOUNT AMARETTO   
159900        67527                     KAHLUA COFFEE LIQUEUR   
159901        28866                             TANQUERAY GIN   
159902        80363  OLE SMOKY BANANA PUDDING CREAM MOONSHINE   
159903        20116                      OLD GRAND DAD 114PRF   

        Bottle Volume (ml)  State Bottle Cost  State Bottle Retail  \
0                      750              18.09                27.14   
1                      750               4.84                 7.26   
2                      750              15.59                23.39   
3                     1000               7.87                11.81   
4                      200               2.50                 3.75   
...                    ...                ...                  ...   
159899                1750               8.07                12.11   
159900                1000              14.81                22.22   
159901                 750              13.50                20.25   
159902                 750              13.00                19.50   
159903                 750              13.84                20.76   

        Bottles Sold  Sale (Do

In [12]:
sale_counts_byDate=iowa_liquor["Date"].value_counts()
# The date field is a DateTime variable. The highest count of liquor sales occurred on Memorial Day weekend in 2020, as well as New Year's Eve in 2020. 
sale_sum_type= iowa_liquor.groupby("Category Name", as_index=False)["Sale (Dollars)"].sum()
sale_sum_type.sort_values("Sale (Dollars)", ascending=False)

Category Name  Sale (Dollars)
24          CANADIAN WHISKIES      2752702.27
15            AMERICAN VODKAS      2632555.67
77  STRAIGHT BOURBON WHISKIES      1600583.13
76                 SPICED RUM      1416282.97
88            WHISKEY LIQUEUR      1262406.28
..                        ...             ...
16                   ANISETTE          598.29
74         SPEARMINT SCHNAPPS          458.92
69                 ROCK & RYE          433.88
31            CREME DE ALMOND          197.77
90      WHITE CREME DE MENTHE          175.21

[92 rows x 2 columns]

In [21]:
iowa_liquor["Date"].isna().sum()
iowa_liquor["Category Name"].isna().sum()
iowa_liquor["Category Name"].count()
133/159771
iowa_liquor["Sale (Dollars)"].sum()
2752702.27/23494950.3

0.11716144255899957

4. Open one of the datasets using Pandas:
    1. `ames_prices.csv`: Housing characteristics and prices
    2. `college_completion.csv`: Public, nonprofit, and for-profit educational institutions, graduation rates, and financial aid
    3. `ForeignGifts_edu.csv`: Monetary and in-kind transfers from foreign entities to U.S. educational institutions
    4. `iowa.csv`: Liquor sales in Iowa, at the transaction level
    5. `metabric.csv`: Cancer patient and outcome data
    6. `mn_police_use_of_force.csv`: Records of physical altercations between Minnessota police and private citizens
    7. `nhanes_data_17_18.csv`: National Health and Nutrition Examination Survey
    8. `tuna.csv`: Yellowfin Tuna Genome (I don't recommend this one; it's just a sequence of G, C, A, T )
    9. `va_procurement.csv`: Public spending by the state of Virginia

5. Pick two or three variables and briefly analyze them
    - Is it a categorical or numeric variable?
    - How many missing values are there? (`df['var'].isna()` and `np.sum()`)
    - If categorical, tabulate the values (`df['var'].value_counts()`) and if numeric, get a summary (`df['var'].describe()`)

Variable 1: Date
    - The date field refers to the purchase date. There were no blank fields in this column. The variable is a Date Time variable. The value_counts show the frequency of each purchase date.
    - The value counts also provide insight into which days have the highest rate of alcohol consumption if you assume that high rates of liquor sales correspond with high rates of liquor consumption. The highest count of liquor sales occurred on Memorial Day weekend in 2020, as well as New Year's Eve in 2020.

Variable 2: Category Name
    - The Category name is a categorical variable describing the type of liquor included in the transaction. There are 133 missing values in this column, accounting for less than 1% of the entries. Interestingly, the liquor type with the highest dollar amount of sales is Canadian Whiskeys. These accounted for nearly 12% of all liquor sales in Iowa across the date range.

6. What are some questions and prediction tools you could create using these data? Who would the stakeholder be for that prediction tool? What practical or ethical questions would it create? What other data would you want, that are not available in your data?

It would be interesting to run ANOVA tests on the zip codes to analyze differences in liquor sales across areas. With an additional data source for population per zip code, you could analyze the areas with the highest liquor sales per capita. 

7. Commit your work to the repo (`git commit -am 'Finish assignment'` at the command line, or use the Git panel in VS Code). Push your work back to Github and submit the link on Canvas in the assignment tab.